build.gradle.kts

```diff
@@ -101,15 +101,32 @@ java {
 }
 
 stutter {
-    isSparse = (findProperty("stutter.sparce")?.toString()?.toBoolean()) ?: true
-    java(8) {
-        compatibleRange("5.0")
-    }
-    java(11) {
-        compatibleRange("5.0")
-    }
-    java(17) {
-        compatibleRange("7.3")
+    sparse.set(providers.gradleProperty("stutter.sparce").map(String::toBoolean).orElse(true))
```

Comment:
Have you checked that locally? The [workaround](https://github.com/gradle-nexus/publish-plugin/blob/7663b49ff9d0ff97fba875ce59172a4fc6b2d226/.github/workflows/java-versions.yml#L28-L34) was causing testing only with the lower supported Gradle version (for a particular Java version)? It could be applied also in the GHA.

Reply:
`gradlew stutterWriteLocks -Pstutter.sparse=false` will generate all the versions
```
java11=6.0.1,6.1.1,6.2.2,6.3,6.4.1,6.5.1,6.6.1,6.7.1,6.8.3,6.9.4,7.0.2,7.1.1,7.2,7.3.3,7.4.2,7.5.1,7.6.1,8.0.2,8.1.1,8.2-rc-1
java17=7.3.3,7.4.2,7.5.1,7.6.1,8.0.2,8.1.1,8.2-rc-1
java8=6.0.1,6.1.1,6.2.2,6.3,6.4.1,6.5.1,6.6.1,6.7.1,6.8.3,6.9.4,7.0.2,7.1.1,7.2,7.3.3,7.4.2,7.5.1,7.6.1,8.0.2,8.1.1,8.2-rc-1
```
vs `gradlew stutterWriteLocks` (with the default sparse=true)
```
java11=6.0.1,6.9.4,7.0.2,7.6.1,8.0.2,8.1.1,8.2-rc-1
java17=7.3.3,7.6.1,8.0.2,8.1.1,8.2-rc-1
java8=6.0.1,6.9.4,7.0.2,7.6.1,8.0.2,8.1.1,8.2-rc-1
```
The strange thing is (which IMO invalidates the usefulness of stutter's `sparse` feature) is that it picks up the lockfile, but doesn't filter it:
```
$ gradlew compatTest --dry-run --no-parallel -Pstutter.sparse=false > sparse-false.txt
$ gradlew compatTest --dry-run --no-parallel -Pstutter.sparse=true > sparse-true.txt
$ gradlew compatTest --dry-run --no-parallel > sparse-default.txt
```
all the above will output the same task list, so even with the longer lockfile.

Reply:
> The strange thing is (which IMO invalidates the usefulness of stutter's sparse feature) is that it picks up the lockfile, but doesn't filter it:

Maybe it would be good to report/ask about that the stutter plugin author?

Reply:
Feel free, since the plugin is under-documented, all behaviour is valid. 🫣

There's one line in the readme that partly explains this behaviour: "lockfile is generated based on the extension configuration" https://github.com/ajoberstar/gradle-stutter#lock-file